src/software/simulated_tests/validation_functions/robot_received_ball_validation.cpp

```diff
@@ -0,0 +1,17 @@
+#include "software/simulated_tests/validation_functions/robot_received_ball_validation.h"
+#include "software/ai/evaluation/robot.h"
+
+void robotReceivedBall(Robot robot, std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield)
```

Comment:
I think we should pass in `Ball` here instead of `world_ptr`

Reply:
I think we determined we can't because only the world_ptr get's updated